src/Vm.sol

```diff
@@ -138,12 +138,50 @@ interface VmSafe {
     // decode the tuple in that order, and thus fail.
+    //
+    // writeJson
+    //
+    // Let's assume we want to write the following JSON to a file:
+    //
+    // { "boolean": true, "number": 342, "object": { "title": "finally json serialization" } }
+    //
+    // ```
+    //  string memory json1 = "some key";
+    //  vm.serializeBool(json1, "boolean", true);
+    //  vm.serializeBool(json1, "number", uint256(342));
+    //  json2 = "some other key";
+    //  string memory output = vm.serializeString(json2, "title", "finally json serialization");
+    //  vm.serialize(json1, "object", output);
+    //  vm.writeJson(json1, "./output/example.json");
+    // ```
+    //  The critical insight is that every invocation of serialization will return the stringified version of the JSON
+    // up to that point. That means we can construct arbitrary JSON objects and then use the return stringified version
+    // to serialize them as values to another JSON object.
+    //
+    //  json1 and json2 are simply keys used by the backend to keep track of the objects. So vm.serializeJson(json1,..)
+    //  will find the object in-memory that is keyed by "some key".   // writeJson
 
+    function serializeBool(string calldata, string calldata, bool) external returns (string memory);
+    function serializeUint(string calldata, string calldata, uint256) external returns (string memory);
+    function serializeInt(string calldata, string calldata, int256) external returns (string memory);
+    function serializeAddress(string calldata, string calldata, address) external returns (string memory);
+    function serializeBytes32(string calldata, string calldata, bytes32) external returns (string memory);
+    function serializeString(string calldata, string calldata, string calldata) external returns (string memory);
+    function serializeBytes(string calldata, string calldata, bytes calldata) external returns (string memory);
+
+    function serializeBool(string calldata, string calldata, bool[] calldata) external returns (string memory);
+    function serializeUint(string calldata, string calldata, uint256[] calldata) external returns (string memory);
+    function serializeInt(string calldata, string calldata, int256[] calldata) external returns (string memory);
+    function serializeAddress(string calldata, string calldata, address[] calldata) external returns (string memory);
+    function serializeBytes32(string calldata, string calldata, bytes32[] calldata) external returns (string memory);
+    function serializeString(string calldata, string calldata, string[] calldata) external returns (string memory);
+    function serializeBytes(string calldata, string calldata, bytes[] calldata) external returns (string memory);
+    function writeJson(string calldata, string calldata) external;
+    function writeJson(string calldata, string calldata, string calldata) external;
```

Comment:
What does `vm.writeJson` do different than `vm.writeFile`, or are they identical? Also can we add some comments explaining the input args here, similar to how we have:

```solidity
// Writes data to file, creating a file if it does not exist, and entirely replacing its contents if it does.
// (path, data) => ()
function writeFile(string calldata, string calldata) external;
```

In particular I'm not clear on what the third `string` argument is